### ISSUE TITLE
Add *optional* watermark to exports

### ISSUE THREAD

**Issue by petehunt** [excalidraw/excalidraw#1312]
Let's add a checkbox that is default *disabled* to add a "Made in Excalidraw" watermark to the lower right corner of exported drawings. We should also include a sentence encouraging users to check it: "Help spread the word about Excalidraw! Check this box to add a watermark to your drawing"

I like the idea of defaulting it to off since I think a lot of people use Excalidraw for its indie vibe, but I'm open to changing the default.

**Comment by vjeux** [comment]
Sounds like a great idea. The first question a lot of people ask is “what’s the illustration done with”. That would help prevent this question.

I feel like a light watermark would be helpful and not super intrusive

**Comment by dwelle** [comment]
I agree it should be off-by-default.

An important consideration is that even if users do opt-in to show it, when other people see it on social media watermarked, they might assume Excalidraw is either freemium (with premium plans for "serious" work), or that we simply have adversarial practices and don't allow to export without watermark which might put them off and they won't even bother investigating further (that's how my mind works).

(Beware of unforeseen consequences)

**Comment by edbentley** [comment]
I'm happy to take this on as my first issue? 😄

**Comment by lipis** [comment]
Go ahead!

**Comment by lipis** [comment]
shall we increase the size of the exported PNG, so it won't interfere in any way with the actual image?! and also it should be easy to get rid of the watermark later on if people want to, just by cropping the image.

**Comment by dwelle** [comment]
Shouldn't we gather consensus on this issue first? AFAIK we haven't gotten back to this since initial post.

**Comment by dwelle** [comment]
One other possibility to soften the potential impact of this is to ensure the watermark explains it's optional:

![image](https://user-images.githubusercontent.com/5153846/78895479-3b1df900-7a6f-11ea-8c02-5944744be8fa.png)

**Comment by lipis** [comment]
No I would just add the checkbox inside the export dialog and just `Made in Excalidraw` should be enough.

**Comment by dwelle** [comment]
The optionality of the watermark (via checkbox) isn't in question here. The question is about the impact on our product/brand perception by new users. If we show watermark, they'll assume we're not really "free", which might have a huge impact (I have no intuition on how big that impact is, but we shouldn't dismiss it outright).

**Comment by lipis** [comment]
I like the idea..

**Comment by edbentley** [comment]
I already made some progress and have a WIP PR: https://github.com/excalidraw/excalidraw/pull/1365. I'll leave it here as a point for discussion, but I'm totally ok with closing it if it's decided this isn't a feature we want

There are a few subtleties to this which are also worth discussing if this is going ahead. For example:

- Should the watermark extend the height of the exported image so it's never interfering with the drawing?
- Font size / opacity / placement of watermark
- Watermark checkbox message (I chose a shorter one than was originally suggested - it'll also need translations)
- Should "Made in Excalidraw" be translated too?

For reference I see [carbon.now.sh](https://carbon.now.sh/) has the same problem. It offers a watermark, though I've never actually seen anyone use it.
